### ISSUE TITLE
Error using skypack ESM module with stimulus-rails

### ISSUE THREAD

**Issue by AxelTheGerman** [afcapel/stimulus-autocomplete#52]
Hi,

I've used stimulus in the past but am new using `stimulus-rails` with ESM modules from skypack.dev.

I'm getting the following error when adding `stimulus-autocomplete` to my project:
```
Failed to autoload controller: autocomplete Error: Unable to resolve specifier 'autocomplete_controller' from http://localhost:3000/assets/stimulus/loaders/autoloader-049…957166e3d004afc916a4c399f9e164045c348d56d6fb01db375ef8acf.js
    throwUnresolved http://localhost:3000/assets/stimulus/libraries/es-module-shims-108c9f5a06bb8dcc1fd6ffac571d699e3bf8d76ba3f09e205a96b98063d61441.js:472
    resolve http://localhost:3000/assets/stimulus/libraries/es-module-shims-108c9f5a06bb8dcc1fd6ffac571d699e3bf8d76ba3f09e205a96b98063d61441.js:468
    importShim$1 http://localhost:3000/assets/stimulus/libraries/es-module-shims-108c9f5a06bb8dcc1fd6ffac571d699e3bf8d76ba3f09e205a96b98063d61441.js:267
    loadController http://localhost:3000/assets/stimulus/loaders/autoloader-049dda1957166e3d004afc916a4c399f9e164045c348d56d6fb01db375ef8acf.js:20
    autoloadControllersWithin http://localhost:3000/assets/stimulus/loaders/autoloader-049dda1957166e3d004afc916a4c399f9e164045c348d56d6fb01db375ef8acf.js:8
    <anonymous> http://localhost:3000/assets/stimulus/loaders/autoloader-049dda1957166e3d004afc916a4c399f9e164045c348d56d6fb01db375ef8acf.js:52
```

My index.js:
```
import { Application } from "stimulus"
import { definitionsFromContext } from "stimulus/webpack-helpers"

import { Autocomplete } from 'https://cdn.skypack.dev/stimulus-autocomplete';

const application = Application.start()
const context = require.context("controllers", true, /_controller\.js$/)
application.load(definitionsFromContext(context))

application.register('autocomplete', Autocomplete)
````

And my HTML view:
```
<div data-controller="autocomplete" data-autocomplete-url="/birds/search">
  <input type="text" data-autocomplete-target="input"/>
  <input type="hidden" name="bird_id" data-autocomplete-target="hidden"/>
  <ul class="list-group" data-autocomplete-target="results"></ul>
</div>
```

Could this be because the `ES Module Entrypoint` is missing for this package? https://www.skypack.dev/view/stimulus-autocomplete

Thank you!

**Comment by drnic** [comment]
Interested in recreating the failing issue on our example repo? Perhaps
create index-es6.html that only loads the skypack files? (I’m new to this
too)--
Dr Nic Williams
+61 437 276 076
twitter @drnic

**Comment by drnic** [comment]
Also, if you think you know what we need to add to this package to fix the issue please create a PR as a suggestion and I'll try to help test it.

**Comment by AxelTheGerman** [comment]
Thank you @drnic I pushed an example repo here https://github.com/AxelTheGerman/stimulus-autocomplete-iisue-52

There are instructions on skypack.dev about ESM modules but as a non-JS-pro it doesn't tell me much. Especially I have no clue how this could be tested without pushing a proper version to skypack.dev
See https://docs.skypack.dev/package-authors/package-checks#esm

For now I'll try to find a simple stimulus controller that (according to skypack.dev) has proper ESM configuration and see if that works out of the box. Will post my findings here.

**Comment by AxelTheGerman** [comment]
Hmm I found https://www.skypack.dev/view/stimulus-clipboard which supposedly has an ESM Entrypoint but getting the same error.

It looks like the ESM Entrypoint is important but even with that I'm still doing something wrong that's unrelated to this library. See discussion here https://discuss.hotwire.dev/t/rails-6-hot-wire-with-asset-pipeline-bootstrap5/1883

If you don't mind we can keep this issue open for a couple of days while I'm gathering more information - or we can close it and I can tag you again to re-open once I do have that information.

**Comment by drnic** [comment]
Yep keep it open. We’re all learning. Thanks for sharing as you go.--
Dr Nic Williams
+61 437 276 076
twitter @drnic

**Comment by AxelTheGerman** [comment]
There are some active discussions in Hotwire's discourse - I managed to actually import `lodash.debounce` which is the only dependency of this project. But haven't been able to import this controller directly yet.

**Comment by afcapel** [comment]
ESM imports are now working in `main`. The [example file shows how to use them](https://github.com/afcapel/stimulus-autocomplete/blob/b95709f33e18935bdef9a297fc320923968f13d9/example/index.html#L10-L25).
